src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonNode/JsonObjectTests.cs

```diff
@@ -969,5 +969,156 @@ public static void LazyInitializationIsThreadSafe()
+            jObject["value"] = JsonValue.Create(10);
+            jObject["array"] = array;
+            jObject["object"] = nestedJsonObj;
+
+            var clone = jObject.DeepClone().AsObject();
+
+            Assert.True(JsonNode.DeepEquals(jObject, clone));
+
+            Assert.Equal(jObject.Count, clone.Count);
+            Assert.Equal(1, clone["One"].GetValue<int>());
+            Assert.Equal(2, clone["Two"].GetValue<int>());
+            Assert.Equal("ABC", clone["String"].GetValue<string>());
+            Assert.True(clone["True"].GetValue<bool>());
+            Assert.False(clone["False"].GetValue<bool>());
+            Assert.Null(clone["Null"]);
+            Assert.Equal(10, clone["value"].GetValue<int>());
+
+            JsonArray clonedArray = clone["array"].AsArray();
+            Assert.Equal(array.Count, clonedArray.Count);
+            Assert.Equal(5, clonedArray[0].GetValue<int>());
+            Assert.Equal(7, clonedArray[1].GetValue<int>());
+
+            JsonObject clonedNestedJObject = clone["object"].AsObject();
+            Assert.Equal(nestedJsonObj.Count, clonedNestedJObject.Count);
+            Assert.Equal(10, clonedNestedJObject["Ten"].GetValue<int>());
+            Assert.Equal("xyz", clonedNestedJObject["Name"].GetValue<string>());
+
+            string originalJson = jObject.ToJsonString();
+            string clonedJson = clone.ToJsonString();
+
+            Assert.Equal(originalJson, clonedJson);
+        }
+
+        [Fact]
+        public static void DeepClone_FromElement()
+        {
+            JsonDocument document = JsonDocument.Parse("{\"One\": 1, \"String\": \"abc\"}");
+            JsonObject jObject = JsonObject.Create(document.RootElement);
+            var clone = jObject.DeepClone().AsObject();
+
+            Assert.True(JsonNode.DeepEquals(jObject, clone));
+            Assert.Equal(1, clone["One"].GetValue<int>());
+            Assert.Equal("abc", clone["String"].GetValue<string>());
+        }
+
+        [Fact]
+        public static void DeepEquals()
+        {
+            var jObject = new JsonObject();
+            jObject["One"] = 1;
+            jObject["array"] = new JsonArray() { "a", "b" };
+
+            var sameJObject = new JsonObject();
+            sameJObject["One"] = 1;
+            sameJObject["array"] = new JsonArray() { "a", "b" };
+
+            Assert.True(JsonNode.DeepEquals(jObject, jObject));
+            Assert.True(JsonNode.DeepEquals(jObject, sameJObject));
+            Assert.True(JsonNode.DeepEquals(sameJObject, jObject));
+
+            Assert.True(JsonNode.DeepEquals(null, null));
+            Assert.False(JsonNode.DeepEquals(jObject, null));
+            Assert.False(JsonNode.DeepEquals(null, jObject));
+
+            var diffJObject = new JsonObject();
+            diffJObject["One"] = 3;
+
+            Assert.False(JsonNode.DeepEquals(diffJObject, jObject));
+            Assert.False(JsonNode.DeepEquals(jObject, diffJObject));
+        }
+
+        [Fact]
```

Comment:
Can you also add a test running `DeepEquals` comparisons between `JsonObject` and `JsonValue` wrapping POCOs or Dictionary types?